tests/test_general_reduction_ops.py

```diff
@@ -295,3 +295,60 @@ def test_accuracy_sum_dim(shape, dim, keepdim, dtype):
     if dim == [] and keepdim:
         _dim = inp.numel()
     gems_assert_close(res_out, ref_out, dtype, reduce_dim=_dim)
+
+
+QUANTILE_Q = (
+    [(0.2, 0.5, 0.8)]
+    if QUICK_MODE
+    else [(0.4), (0.0, 0.2, 0.5, 0.8, 1.0), (0.662, 0.8, 0.104, 0.99, 0.347)]
+)
+QUANTILE_INTERPOLATION = (
+    ["linear"] if QUICK_MODE else ["linear", "lower", "higher", "nearest", "midpoint"]
+)
+
+
+@pytest.mark.quantile
+@pytest.mark.parametrize("shape", REDUCTION_SMALL_SHAPES)
+@pytest.mark.parametrize("dtype", [torch.float32])
+@pytest.mark.parametrize("q", QUANTILE_Q)
+@pytest.mark.parametrize("interpolation", QUANTILE_INTERPOLATION)
+def test_accuracy_quantile_without_dim(shape, dtype, q, interpolation):
+    inp = torch.randn(shape, dtype=dtype, device="cuda")
+    ref_inp = to_reference(inp, True)
+    q = torch.tensor(q, device=inp.device)
+    ref_q = to_reference(q, True)
+
+    ref_out = torch.quantile(ref_inp, ref_q, interpolation=interpolation)
+    with flag_gems.use_gems():
+        res_out = torch.quantile(inp, q, interpolation=interpolation)
+
+    gems_assert_close(res_out, ref_out, dtype, reduce_dim=inp.numel())
+
+
+@pytest.mark.quantile
+@pytest.mark.parametrize("shape", REDUCTION_SMALL_SHAPES)
+@pytest.mark.parametrize("keepdim, dim", KEEPDIM_DIM)
+@pytest.mark.parametrize("dtype", [torch.float32])
+@pytest.mark.parametrize("q", QUANTILE_Q)
+@pytest.mark.parametrize("interpolation", QUANTILE_INTERPOLATION)
+def test_accuracy_quantile_dim(shape, dim, keepdim, dtype, q, interpolation):
+    inp = torch.randn(shape, dtype=dtype, device="cuda")
+    ref_inp = to_reference(inp, True)
```

Comment:
You should change this to use to_reference(inp) for the float32 case to ensure consistency in the dtype. Additionally, consider adding more comprehensive test coverage for various dtypes to ensure the robustness of the implementation.